### PR TITLE
[sensors][android] Convert expo-sensors to Kotlin

### PR DESCRIPTION
**Pull request by ixf**
# Why

Java packages are being migrated to Kotlin.

# How

Autoconvert + style changes and type fixes (mostly related to nullability). Couple functions have been turned to properties (where it makes sense). Couple properties are now lateinit or lazy initialized. 

# Test Plan

All sensors on the SensorsScreen in NCL work fine

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BaseSensorModule.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    // Unregister from old UIManager
+    if (this::moduleRegistry.isInitialized && moduleRegistry.getModule(UIManager::class.java) != null) {
+      moduleRegistry.getModule(UIManager::class.java).unregisterLifecycleEventListener(this)
+    }
+    this.moduleRegistry = moduleRegistry
+
+    // Register to new UIManager
+    if (moduleRegistry.getModule(UIManager::class.java) != null) {
+      moduleRegistry.getModule(UIManager::class.java).registerLifecycleEventListener(this)
+    }
```
Comment (bbarthec) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BaseSensorModule.kt:39]:
```suggestion
    moduleRegistry.getModule(UIManager::class.java)?.registerLifecycleEventListener(this)
```

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BaseSensorModule.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  protected abstract val eventName: String
+  protected abstract fun getSensorService(): SensorServiceInterface
+  protected abstract fun eventToMap(sensorEvent: SensorEvent): Bundle
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    // Unregister from old UIManager
+    if (this::moduleRegistry.isInitialized && moduleRegistry.getModule(UIManager::class.java) != null) {
+      moduleRegistry.getModule(UIManager::class.java).unregisterLifecycleEventListener(this)
+    }
+    this.moduleRegistry = moduleRegistry
```
Comment (bbarthec) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BaseSensorModule.kt:34]: Let's clean it up a bit.
We can try to unregister the `lifecycleEventListener` even if it was not previously registered. These `if` conditions don't guard before anything to be honest 😅 
```suggestion
    this.moduleRegistry = moduleRegistry

    // Unregister from old UIManager
    moduleRegistry.getModule(UIManager::class.java)?.unregisterLifecycleEventListener(this)
```

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/DeviceMotionModule.kt [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  private var mUpdateInterval = 1.0f / 60.0f
+  private val mRotationMatrix = FloatArray(9)
+  private val mRotationResult = FloatArray(3)
+  private var mAccelerationEvent: SensorEvent? = null
+  private var mAccelerationIncludingGravityEvent: SensorEvent? = null
+  private var mRotationEvent: SensorEvent? = null
+  private var mRotationRateEvent: SensorEvent? = null
+  private var mGravityEvent: SensorEvent? = null
+  private lateinit var mServiceSubscriptions: MutableList<SensorServiceSubscriptionInterface>
+  private lateinit var mUiManager: UIManager
```
Comment (bbarthec) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/DeviceMotionModule.kt:37]: We tend not to decapitalize abbreviations like this one 😉 
```suggestion
  private lateinit var mUIManager: UIManager
```

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/BaseService.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  var experienceIsForegrounded = false
+    private set
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    mModuleRegistry = moduleRegistry
+
+    // Register to new UIManager
+    if (mModuleRegistry.getModule(UIManager::class.java) != null) {
+      mModuleRegistry.getModule(UIManager::class.java).registerLifecycleEventListener(this)
+    }
```
Comment (bbarthec) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/BaseService.kt:22]:
```suggestion
    mModuleRegistry.getModule(UIManager::class.java)?.registerLifecycleEventListener(this)
```

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/BaseService.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  override fun onDestroy() {
+    // Unregister from old UIManager
+    if (this::mModuleRegistry.isInitialized && mModuleRegistry.getModule(UIManager::class.java) != null) {
+      mModuleRegistry.getModule(UIManager::class.java).unregisterLifecycleEventListener(this)
+    }
```
Comment (bbarthec) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/BaseService.kt:29]:
```suggestion
    if (this::mModuleRegistry.isInitialized) {
      mModuleRegistry.getModule(UIManager::class.java)?.unregisterLifecycleEventListener(this)
    }
```

**Review by bbarthec — COMMENTED** [review]
Looks solid and safe 💪 
Thank you for this work! 
I proposed just a few small suggestions 😉

**Comment by ixf** [comment]
Updates the imports to work after this change: https://github.com/expo/expo/pull/13703

**Review by bbarthec — APPROVED** [review]
Once the CI completes I'm going to merge this 💪 
Good work 👍

**Pull request merged**